adam/perception/perception_graph_nodes.py

```diff
@@ -136,8 +136,25 @@ class StrokeGNNRecognitionNode(GraphNode):
     def dot_label(self):
         return (
             f"StrokeGNNRecognitionNode(object_recognized={self.object_recognized}, "
-            f"confidence={self.confidence})"
+            f"confidence={self.confidence:.4f})"
         )
 
     def __str__(self) -> str:
         return f"StrokeGNNRecognized(recognized object={self.object_recognized} ({self.confidence:.2f}))"
+
+
+@attrs(frozen=True, slots=True, eq=False)
+class JointPointNode(GraphNode):
+    world_coord: Point = attrib(validator=instance_of(Point))
+    scene_xyd_coord: DepthPoint = attrib(validator=instance_of(DepthPoint))
+    temporal_index: int = attrib(validator=instance_of(int))
+    confidence: float = attrib(validator=instance_of(float))
+
+    def dot_label(self) -> str:
+        return (
+            f"JointPointNode(world_coord={self.world_coord}, scene_xyd_coord={self.scene_xyd_coord},"
+            f"confidence={self.confidence:.4f}, temporal_index={self.temporal_index})"
+        )
+
+    def __str__(self) -> str:
+        return f"JointPointNode([{self.temporal_index}], world_coord={self.world_coord}, scene_zyd_coord={self.scene_xyd_coord} ({self.confidence:.2f}))"
```

Comment:
Ah, similar typo here, missed it the first time.
```suggestion
        return f"JointPointNode([{self.temporal_index}], world_coord={self.world_coord}, scene_xyd_coord={self.scene_xyd_coord} ({self.confidence:.2f}))"
```